assets/js/blocks/cart-checkout/mini-cart/edit.tsx

```diff
@@ -1,31 +1,105 @@
 /**
  * External dependencies
  */
-import { useBlockProps } from '@wordpress/block-editor';
+import {
+	InspectorControls,
+	useBlockProps,
+	getColorClassName,
+} from '@wordpress/block-editor';
 import type { ReactElement } from 'react';
 import { formatPrice } from '@woocommerce/price-format';
 import { CartCheckoutCompatibilityNotice } from '@woocommerce/editor-components/compatibility-notices';
+import { PanelBody, ToggleControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
 
 /**
  * Internal dependencies
  */
 import QuantityBadge from './quantity-badge';
 
-const MiniCartBlock = (): ReactElement => {
+interface Attributes {
+	isInitiallyOpen?: boolean;
+	transparentButton: boolean;
+	colorClassNames?: string;
+	style?: Record< string, Record< string, string > >;
+}
+
+interface Props {
+	attributes: Attributes;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
+}
+
+const MiniCartBlock = ( {
+	attributes,
+	setAttributes,
+}: Props ): ReactElement => {
+	const { transparentButton, backgroundColor, textColor, style } = attributes;
 	const blockProps = useBlockProps( {
-		className: 'wc-block-mini-cart',
+		className: classnames(
+			'wc-block-mini-cart wp-block-woocommerce-mini-cart',
+			{
+				'is-transparent': transparentButton,
+			}
+		),
 	} );
 
+	const backgroundClass = getColorClassName(
```

Comment:
Also looking at the Table block, I discovered that there is a [`__experimentalUseColorProps`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/table/edit.js#L17) that makes it easier to access the [class names](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/table/edit.js#L478) and [styles](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/table/edit.js#L482). Given that it's still experimental, I think it's ok if we leave the code in this PR as-is, but wondering if we should add a `@todo` comment so we remember to replace this implementation with `useColorProps()` once it's no longer experimental.